pkg/cmd/operator/webhooks_test.go

```diff
@@ -0,0 +1,302 @@
+
+			err := test.WebhookOptions.Complete()
+			if err != nil {
+				t.Fatalf("can't complete WebhookOptions: %v", err)
+			}
+
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				err := test.WebhookOptions.run(ctx, genericclioptions.IOStreams{})
+				if !reflect.DeepEqual(test.ExpectedError, err) {
+					t.Errorf("expected and actual errors differ: %s", cmp.Diff(test.ExpectedError, err))
+				}
+			}()
+		})
+	}
+}
+
+func TestRunWebhookServerCertReload(t *testing.T) {
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dir := t.TempDir()
+	tlsCertFile := dir + "/tls.crt"
+	tlsKeyFile := dir + "/tls.key"
+
+	cc, err := newCertificateCreator()
+	if err != nil {
+		t.Fatalf("failed to create CertificateCreator: %v", err)
+	}
+
+	initialCertSerialNumber := big.NewInt(1)
+	encodedCert, err := cc.generateEncodedCertificate(initialCertSerialNumber)
+	if err != nil {
+		t.Fatalf("failed to create new encoded certificate: %v", err)
+	}
+	if err = createFileWithContent(tlsCertFile, encodedCert.cert); err != nil {
+		t.Fatalf("failed to create a file: %v", err)
+	}
+	if err = createFileWithContent(tlsKeyFile, encodedCert.privateKey); err != nil {
+		t.Fatalf("failed to create a file: %v", err)
+	}
+
+	wo := WebhookOptions{
+		TLSCertFile:                    tlsCertFile,
+		TLSKeyFile:                     tlsKeyFile,
+		Port:                           0,
+		InsecureGenerateLocalhostCerts: false,
+	}
+	err = wo.Complete()
+	if err != nil {
+		t.Fatalf("failed to complete WebhookOptions: %v", err)
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		err := wo.run(ctx, genericclioptions.IOStreams{})
+		if err != nil {
+			t.Errorf("can't run webhook server: %v", err)
+		}
+	}()
+
+	certpool := x509.NewCertPool()
+	certpool.AppendCertsFromPEM(cc.encodedCA)
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: certpool,
+			},
+		},
+	}
+
+	wo.assignedPort.c.L.Lock()
+	for wo.assignedPort.value == nil {
+		wo.assignedPort.c.Wait()
+	}
+	addr := fmt.Sprintf("https://localhost:%d/readyz", *wo.assignedPort.value)
+	wo.assignedPort.c.L.Unlock()
+
+	observeCertificate := func(expectedCertificateSerialNumber *big.Int) func() (bool, error) {
+		return func() (bool, error) {
+			resp, err := client.Get(addr)
+			if err != nil {
+				t.Logf("can't make a GET request: %v", err)
+				return false, nil
+			}
+
+			if !reflect.DeepEqual(expectedCertificateSerialNumber, resp.TLS.PeerCertificates[0].SerialNumber) {
+				t.Logf("serial numbers differ: expected %s, actual %s", expectedCertificateSerialNumber, resp.TLS.PeerCertificates[0].SerialNumber)
+				return false, nil
+			}
+
+			return true, nil
+		}
+	}
+
+	err = wait.PollImmediate(pollInterval, pollTimeout, observeCertificate(initialCertSerialNumber))
+	if err != nil {
+		t.Fatalf("failed to observe the initial certificate: %v", err)
+	}
+
+	replaceCertSerialNumber := big.NewInt(2)
```

Comment:
nit: guess the name should be `replacedCertSerialNumber` but a better naming might be `initialCert...` and `updatedCert...`